### PR TITLE
Introduce 'archived page' banner

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR adds a banner to the top of a page when it's archived.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202511881285731/f) _(internal)_

## 🔍 What does this change?

Added a `PageNotificationBanner` component to the layout, managed by the `PageNotiticationBannerContext` (similar to how the sidebar is implemented).

## ❓ How to test this?

Try to archive a combination of pages/subpages and check if the banner is displayed when inside the page that was just archived. Also, click the `restore` button and check if the page is back on the account pages list.

## 📹 Demo

banner

### CODE REVIEW

File: packages/hash/frontend/src/shared/layout/layout-with-sidebar/notification-banner.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <Box
+      sx={({ palette }) => ({
+        color: palette.common.white,
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        width: 1,
+        background: "#EB5757",
```
Comment (nonparibus) [inline packages/hash/frontend/src/shared/layout/layout-with-sidebar/notification-banner.tsx:21]: Please avoid hard-coding colors in files. We have red 50/60 theme colors that would work here. See https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=828%3A67111

[the remark this answers is not in the document]
Comment (luisbettencourt) [reply packages/hash/frontend/src/shared/layout/layout-with-sidebar/notification-banner.tsx]: My bad, thought I would get back to it butforgot to. Addressed in f2ca1bf4a3b685a8d9e072bb9550dff18db6e0ed

Comment (luisbettencourt) [reply packages/hash/frontend/src/shared/layout/layout-with-sidebar/notification-banner.tsx]: Addressed in f2ca1bf4a3b685a8d9e072bb9550dff18db6e0ed

File: packages/hash/frontend/src/components/hooks/useArchivePage.ts [hunk 2]
```diff
@@ -1,45 +1,69 @@
...
-      {
-        query: getAccountPages,
-        variables: { accountId },
-      },
-    ],
-  });
+  >(updatePage);
+
+  const getRefecthQueries = useMemo(
+    () => (accountId: string, pageEntityId: string) =>
```
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply packages/hash/frontend/src/components/hooks/useArchivePage.ts]: Addressed in e43cbab4506ae16e590867ed95a5ad84d48f5da4

**Review by nathggns — DISMISSED** [review]
Great work Luís – thanks for bearing with changes!

**Review by nonparibus — APPROVED** [review]

**Comment by luisbettencourt** [comment]
I added 4820515ca52a4d69cfcb5cf2e221a056a8dd1727 to prevent the warning we were getting when we archived all pages in the account pages list (thus making it empty):

```
Cache data may be lost when replacing the accountPages field of a Query object.

To address this problem (which is not a bug in Apollo Client), define a custom merge function for the Query.accountPages field, so InMemoryCache can safely merge these objects:

  existing: [{"__typename":"Page","entityId":"cd24d9f3-1100-40a8-b316-1c6b316f4616","properties":{"__typename":"PageProperties","title":"Untitled"},"parentPageEntityId":null}]
  incoming: []

For more information about these options, please refer to the documentation:

  * Ensuring entity objects have IDs: https://go.apollo.dev/c/generating-unique-identifiers
  * Defining custom merge functions: https://go.apollo.dev/c/merging-non-normalized-objects
```

But it may be unnecessary as the behavior is the same. What do you think @nathggns ?

**Comment by CiaranMn** [comment]
> I added https://github.com/hashintel/hash/commit/4820515ca52a4d69cfcb5cf2e221a056a8dd1727 to prevent the warning we were getting when we archived all pages in the account pages list (thus making it empty):

I think this is fine. 

You've also [added](https://github.com/hashintel/hash/commit/01114df543a6e99d14e93182ee37caaa670caac3) `merge: true` to `Page` – was this to address the warning about `PageProperties` not having an id? In our current usage, your approach works, because we never request `PageProperties` independently of `Page`. 

If we _did_ (e.g. we had some `getPageProperties` query which returned only a `PageProperties` object), we'd want to make sure `PageProperties` had a unique identifier to allow it to be cached properly, and so that the same `PageProperties` would be identified and the cache updated whether it was delivered the child of a `Page` or not. [Here's one way](https://github.com/hashintel/hash/compare/lb/archivation-banner...cm/archivation-banner-cachekeys) of doing that. As I say, I don't think you need to change your PR, because it's fine for us to assume that we only ever get `PageProperties` as the child of `Page`, and the merge strategy works.

**Comment by luisbettencourt** [comment]
> > I added [4820515](https://github.com/hashintel/hash/commit/4820515ca52a4d69cfcb5cf2e221a056a8dd1727) to prevent the warning we were getting when we archived all pages in the account pages list (thus making it empty):
> 
> I think this is fine.
> 
> You've also [added](https://github.com/hashintel/hash/commit/01114df543a6e99d14e93182ee37caaa670caac3) `merge: true` to `Page` – was this to address the warning about `PageProperties` not having an id? In our current usage, your approach works, because we never request `PageProperties` independently of `Page`.
> 
> If we _did_ (e.g. we had some `getPageProperties` query which returned only a `PageProperties` object), we'd want to make sure `PageProperties` had a unique identifier to allow it to be cached properly, and so that the same `PageProperties` would be identified and the cache updated whether it was delivered the child of a `Page` or not. [Here's one way](https://github.com/hashintel/hash/compare/lb/archivation-banner...cm/archivation-banner-cachekeys) of doing that. As I say, I don't think you need to change your PR, because it's fine for us to assume that we only ever get `PageProperties` as the child of `Page`, and the merge strategy works.

Thanks Ciaran, that was it. I like that solution better so I'll merge that branch with this one.

**Comment by nathggns** [comment]
Reviewing this now

**Review by nathggns — APPROVED** [review]

**Comment by nathggns** [comment]
@luisbettencourt well done on this. It's changed a fair bit since the first inception and I hope you're very happy with where it's ended up! Well done on working through the improvements.

**Pull request merged**